base/secretbuffer.jl

```diff
@@ -106,6 +106,18 @@ end
 const _sb_hash = UInt === UInt32 ? 0x111c0925 : 0xb06061e370557428
 hash(s::SecretBuffer, h::UInt) = hash(_sb_hash, h)
 
+copy(s::SecretBuffer) = copy!(SecretBuffer(sizehint=length(s.data)), s)
```

Comment:
Why not use `copy(s::SecretBuffer) = copy!(SecretBuffer(sizehint=s.size, s)`, with assorted changes in `copy!`?

Reply:
Yes, I considered that, but this seemed simpler and more obviously correct.  Anywhere else and I'd do that optimization readily, but here there is a strong value to simple code.